### ISSUE TITLE
Failed to reproduce

### ISSUE THREAD

**Issue by bx-li** [ZM7/HGLS#2]
Thank your code for this paper, but I failed to reproduce .Can someone tell me where the results in the paper came from? I ran 500 epochs on ICEWS14s and the results are shown in the following figure, but did not achieve the expected results. I slightly modified the code so that ICEWS14s can also be loaded from folder
![图片](https://github.com/CRIPAC-DIG/HGLS/assets/137079616/fa6fe124-c837-4a71-a62f-c26b80a7a6e4)

**Comment by hummingg** [comment]
I got the expected results on ICEWS14s at  about epoch 11. 
![image](https://github.com/CRIPAC-DIG/HGLS/assets/25954059/ad91ab17-af2a-4f07-bcea-3c6330396c66)
![image](https://github.com/CRIPAC-DIG/HGLS/assets/25954059/20617826-4e36-46cd-a88b-0f24fdc7d087)

**Comment by binchen4110** [comment]
you can try other dataset, not only ICEWS14s. And I fail to  run other datasets rather than ICEWS14s.

**Comment by bx-li** [comment]
> 我在 ICEWS14 上获得了大约 11 元的预期结果。 ![image](https://private-user-images.githubusercontent.com/25954059/297052424-ad91ab17-af2a-4f07-bcea-3c6330396c66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0MTEyNjksIm5iZiI6MTcwNTQxMDk2OSwicGF0aCI6Ii8yNTk1NDA1OS8yOTcwNTI0MjQtYWQ5MWFiMTctYWYyYS00ZjA3LWJjZWEtM2M2MzMwMzk2YzY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDEzMTYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiZDQzMzc4YjlmMTU3NTQ4OTJhZjVhMTQ2MjNkYzc1NTU4MWU1ZDRlN2Q3NWJiZTZhYTkwYWQyZjlhMDMzNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yfazFqw4QqVlZQ_LLtdRgysVAlQp7UpU4LebNwQ6hAs) ![image](https://private-user-images.githubusercontent.com/25954059/297055423-20617826-4e36-46cd-a88b-0f24fdc7d087.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0MTEyNjksIm5iZiI6MTcwNTQxMDk2OSwicGF0aCI6Ii8yNTk1NDA1OS8yOTcwNTU0MjMtMjA2MTc4MjYtNGUzNi00NmNkLWE4OGItMGYyNGZkYzdkMDg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDEzMTYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2MTk5NGNlN2M3NzRjMDk0NTFiMjIxM2Y3M2JkMjA0YmEzZjJlNjkxYTQ4YThkYzU4NDNjNTBiMTJlZGU3MDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AULJbpolyI_atBlkdMokp3q8p43KDzKnuWmuB1vd4CU)

Shouldn't the model be tested after its loss converges? The loss is still decreasing here. If we choose the 10th epoch as the result, why do we still set a default training of 500 epochs

**Comment by bx-li** [comment]
> 您可以尝试其他数据集，而不仅仅是 ICEWS14s。而且我无法运行其他数据集而不是 ICEWS14。



> you can try other dataset, not only ICEWS14s. And I fail to run other datasets rather than ICEWS14s.

I am also unable to run on the 05-15 and GDELT datasets, but my confusion is whether the results in the paper were obtained by testing Mrr after 500 epochs_ The value of the raw parameter.

**Comment by hummingg** [comment]
> you can try other dataset, not only ICEWS14s. And I fail to run other datasets rather than ICEWS14s.

You're right. On ICEWS05-15, I found "val_Mrr_raw:0.46"(expected) in validating, but didn't find it in testing.
I added some codes to record the best model.
```python
# main.py
        ...
        if val_result[1][0] > best_val_Mrr_raw:
            best_val_Mrr_raw = val_result[1][0]
            torch.save(model.state_dict(), save_model_path)
            print('#'*10, 'better model found')

        print('\tStart testing: ', format_now())
        ...
```
